src/clothing_mod.cpp

```diff
@@ -118,3 +130,21 @@ const std::vector<clothing_mod> &clothing_mods::get_all()
 {
     return all_clothing_mods.get_all();
 }
+
+const std::vector<clothing_mod> &clothing_mods::get_all_with( clothing_mod_type type )
+{
+    const auto iter = clothing_mods_by_type.find( type );
+    if( iter != clothing_mods_by_type.end() ) {
+        return iter->second;
+    } else {
+        // Build cache
+        std::vector<clothing_mod> *list = new std::vector<clothing_mod> {};
```

Comment:
Why allocate via `new`? And you're not even deleting it, so it leaks memory.

Reply:
it's my mistake, i will fix it.